### ISSUE TITLE
Always tries to connect to 127.0.0.1:8500

### ISSUE THREAD

**Issue by devth** [Cimpress-MCP/git2consul#65]
This seems to be a recent problem. When I run git2consul, regardless of my `endpoint` config, it's trying to hit `127.0.0.1:8500` for some reason:

```json
"err": {"message":"connect ECONNREFUSED 127.0.0.1:8500", ...}
```

I'm running latest. Verified `global.endpoint` is correct by logging it.

**Comment by cleung2010** [comment]
The only reference (apart from utils/config_seeder.js) on `127.0.0.1` is on [#L12](https://github.com/Cimpress-MCP/git2consul/blob/545b4e5920c2204c2e324102c203de34fe145c0f/lib/index.js#L12). How are you passing in the endpoint option, with `-e`, `--endpoint` flag or with the environment variable `CONSUL_ENDPOINT`?

**Comment by devth** [comment]
I've tried all 3. And global.endpoint is correct.

**Comment by devth** [comment]
Here's the rest of the `err`:

```json
{"stack":"Error: connect ECONNREFUSED 127.0.0.1:8500\n    at Object.exports._errnoException (util.js:874:11)\n    at exports._exceptionWithHostPort (util.js:897:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:8500","time":"2016-03-21T15:30:23.604Z","v":0}
```

Maybe the problem is on my end. Still investigating...

**Comment by devth** [comment]
127.0.0.1 is coming from the consul lib. When it's required [here](https://github.com/Cimpress-MCP/git2consul/blob/1d88bcee25af055328de139cd88f8f0f2b58b1b7/lib/config_reader.js#L1), `global` is undefined.

**Comment by cleung2010** [comment]
Is it because index.js is requiring config_seeder.js and config_reader.js before the variables are being set?

**Comment by devth** [comment]
Yes. Happened in https://github.com/Cimpress-MCP/git2consul/commit/d89f571fb35c9aae81f555ecce068523c81f6b25

**Comment by cleung2010** [comment]
Going to issue a patch for that, thanks for the report!

**Comment by devth** [comment]
Great, thanks!

**Comment by cleung2010** [comment]
This should be patched on v0.12.8 now.

**Comment by devth** [comment]
@cleung2010 thanks, works now!
